.github/scripts/sync-templates.js

```diff
@@ -0,0 +1,37 @@
+const path = require('path')
+const semver = require('semver')
+const fse = require('fs-extra')
+
+const templateDir = path.join(__dirname, "../../templates");
+
+function recursivelyListPackageJsonFilePath(dir, list = []) {
+    const arr = fse.readdirSync(dir);
+    arr.forEach(function (item) {
+        if (item === "node_modules") return list;
+        const fullpath = path.join(dir, item);
+        const stats = fse.statSync(fullpath);
+        if (stats.isDirectory()) {
+            recursivelyListPackageJsonFilePath(fullpath, list);
+        } else if (item === "package.json") {
+            list.push(fullpath);
+        }
```

Comment:
seems function (item) doesn't always have return value. Only return list when it is "node_modules".

Reply:
Besides, the code here to iterate the path to find template package.json is not very readable. Can we delegate this to Template? Template package needs tell CD what are package.json needs to sync? As for currenly, it is only several templates needs to handle right?

Reply:
this function updated.
I think it's better to go through the whole dir and get all the package.json file.
And this method already works fine